### PR TITLE
Office endpoints

### LINKED ISSUE #31: Office endpoint

### PR DESCRIPTION
**Pull request by SandroShera**
Closes #31

### CODE REVIEW

File: src/Api/Controllers/OfficesController.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            return await _officeService.GetOffices(filterModel);
+        }
+
+        // GET api/<OfficesController>/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<OfficeViewModel>> GetOfficeById(Guid id)
+        {
+            var result = await _officeService.GetOfficeById(id);
+
+            if (result == null) return new NotFoundResult();
```
Comment (khamza85) [inline src/Api/Controllers/OfficesController.cs:39]: You may want to use `NoContent()` instead as per Rest API convention

Comment (khamza85) [reply src/Api/Controllers/OfficesController.cs:39]: Or just make the return type `Task<OfficeViewModel?>` and return immediate result of _officeService.GetOfficeById(id);

**Review by Andrei-Ha — APPROVED** [review]

File: src/Api/Controllers/OfficesController.cs [hunk 2]
```diff
@@ -1,5 +1,5 @@
-﻿using Exadel.OfficeBooking.Api.DTO.officeDto;
-using Exadel.OfficeBooking.Api.DTO.OfficeDto;
+﻿using Exadel.OfficeBooking.Api.DTO;
+using Exadel.OfficeBooking.Api.DTO;
```
Comment (apfpo) [inline src/Api/Controllers/OfficesController.cs:2]: delete second line

**Review by apfpo — APPROVED** [review]

File: src/Api/Controllers/OfficesController.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        private readonly IOfficeService _officeService;
+        public OfficesController(IOfficeService officeService)
+        {
+            _officeService = officeService;
+        }
+        // GET: api/<OfficesController>
+        [HttpGet]
+        [Produces("application/json")]
+        public async Task<ActionResult<List<OfficeDto>>> GetOffices([FromQuery]OfficeFilterDto filterModel)
```
Comment (khamza85) [inline src/Api/Controllers/OfficesController.cs:29]: Return type can be `Task<OfficeDto[]>` and thus you can `return _officeService.GetOffices(filterModel);`.

File: src/Api/Services/OfficeService.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    {
+        private readonly AppDbContext _context;
+
+        public OfficeService(AppDbContext context)
+        {
+            _context = context;
+        }
+        public async Task<List<OfficeDto>> GetOffices(OfficeFilterDto filterModel)
+        {
+            var offices = _context.Offices.AsQueryable();
```
Comment (khamza85) [inline src/Api/Services/OfficeService.cs:24]: Same as #r803849772

**Review by Bazilis — APPROVED** [review]

**Pull request merged**